DB/api/db.py

```diff
@@ -66,6 +75,80 @@ def query(parsed_url):
 		return format_data(header, results, json)
 
 
+def generate_total_energy(device_id, start_time, end_time, channel):
+	"""
+	Returns total energy for a particular "channel" (device in table data_raw),
+	over a specified time period
+
+	:param device_id: The serial number of the device in question
+	:param start_time: The start of the time range for which to query for data
+	:param end_time: The end of the time range for which to query for data
+	:param channel: channel filter
+	"""
+
+	# Initialize parameter list and WHERE clause
+	params = [device_id]
+	query = "SELECT CAST(data as DECIMAL) / 36e5 FROM " + TABLE + " as raw WHERE serial = %s AND type = 'P'"
+
+	# Generate WHERE clause
+	if start_time and end_time and channel:
+		query += " AND device = %s AND (time = to_timestamp(%s) OR time = to_timestamp(%s));"
```

Comment:
I thought about that as well, I guess this is just a first pass at the problem. thought of this
```
select data, to_timestamp(%s) - time difference from data_raw 
where
    to_timestamp(%s) > time 
order by abs(extract(epoch from passed_ts - ts_column))
limit 1;
```

but this way id have to do two queries